datumaro/plugins/data_formats/camvid.py

```diff
@@ -350,9 +350,14 @@ def apply(self):
                 else:
                     segm_list[item.id] = (image_path, "")
 
+                self._check_hash_key_existence(item)
+
             self.save_segm_lists(subset_name, segm_list)
         self.save_label_map()
 
+        if self._save_hashkey_meta:
+            self._save_hashkey_file(self._save_dir)
+
```

Comment:
Please not adding this to everywhere but just modify `Exporter::apply` as follows.
https://github.com/openvinotoolkit/datumaro/blob/cdc2df6cd3f1898e6b467a49a78b88f71945d1ad/datumaro/components/exporter.py#L164-L165

```python
    def apply(self):
         if self._save_hashkey_meta:
                self._save_hashkey_file(self._save_dir)
         self._apply_impl()

    def _apply_impl(self):
         raise NotImplementedError("Should be implemented in a subclass") 
```

In the current way, we have to care about this again and again for data formats which will be added in the future.

Reply:
I updated this in [4113433](https://github.com/openvinotoolkit/datumaro/commit/411343324cd5f318a511b27f9969e2f69b6dadc0).